<launch>
    <arg name="topic_sub_point_cloud" default="steer_bot/rslidar_back"/>
    <arg name="topic_point_ground" default="$(arg topic_sub_point_cloud)/filtered_points_ground"/>
    <arg name="topic_point_no_ground" default="$(arg topic_sub_point_cloud)/filtered_points_no_ground"/>
    <arg name="clip_height" default="0.2"/>
    <arg name="min_distance" default="0.1"/>
    <arg name="radial_divider_angle" default="0.18"/>
    <arg name="sensor_height" default="1.2"/>
    <arg name="concentric_div_dist" default="0.1"/>
    <arg name="threshold_min_heigh" default="0.05"/>
    <arg name="slope_local_max" default="8"/>
    <arg name="slope_general_max" default="4"/>
    <arg name="reclass_distance_threshold" default="0.2"/>

    <node pkg="ray_filter_ground" type="ray_filter_ground_node" name="ray_filter_ground_back" output="screen">
        <param name="topic_sub_point_cloud" value="$(arg topic_sub_point_cloud)"/>
        <param name="topic_point_ground" value="$(arg topic_point_ground)"/>
        <param name="topic_point_no_ground" value="$(arg topic_point_no_ground)"/>
        <param name="clip_height" value="$(arg clip_height)"/>
        <param name="min_distance" value="$(arg min_distance)"/>
        <param name="radial_divider_angle" value="$(arg radial_divider_angle)"/>
        <param name="sensor_height" value="$(arg sensor_height)"/>
        <param name="concentric_div_dist" value="$(arg concentric_div_dist)"/>
        <param name="threshold_min_heigh" value="$(arg threshold_min_heigh)"/>
        <param name="slope_local_max" value="$(arg slope_local_max)"/>
        <param name="slope_general_max" value="$(arg slope_general_max)"/>
        <param name="reclass_distance_threshold" value="$(arg reclass_distance_threshold)"/>
    </node>
</launch>